nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/objects/CustomJSONObject.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *         Vladimir Pasquier <vpasquier@nuxeo.com>
+ */
+package org.nuxeo.client.test.objects;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "userId", "token" })
+public class CustomJSONObject {
+
+    @JsonProperty("userId")
+    private String userId;
+
+    @JsonProperty("token")
+    private String token;
+
+    @JsonIgnore
+    private Map<String, Object> additionalProperties = new HashMap<String, Object>();
+
+    @JsonProperty("userId")
+    public String getUserId() {
+        return userId;
+    }
+
+    @JsonProperty("userId")
+    public void setUserId(String userId) {
+        this.userId = userId;
+    }
+
+    @JsonProperty("token")
+    public String getToken() {
+        return token;
+    }
```

Comment:
Duplicate annotation. The annotation `@JsonProperty("token")` has already been added in the field of the class. According to the Javadoc of [JsonIgnore (Jackson-annotations 2.7.0 API)][1]:

> In addition, starting with Jackson 1.9, if this is the only annotation associated with a property, it will also cause cause the whole property to be ignored: that is, if setter has this annotation and getter has no annotations, getter is also effectively ignored.

So it can be safely removed, and same for all the rest.

[1]: http://fasterxml.github.io/jackson-annotations/javadoc/2.8/com/fasterxml/jackson/annotation/JsonIgnore.html